### DISCUSSION TITLE
Help, how to exit a higher level state

### DISCUSSION THREAD

**Discussion by Gronk95** [StateSmith/StateSmith#234] [Q&A] [answered]
![VECU_SM2 drawio](https://github.com/StateSmith/StateSmith/assets/140708711/e923a3e5-b368-4777-a3de-85a4fa4561ed)

Hi,
I have a problem in trying to exit a higher level state. In my diagram I have a COLD_BURN state which is composed of a number of sub-states. Each of the sub-states needs to be executed in turn, i.e. we wait for the burn to begin, do the burn and shut down the burn. Then we should transition back to STANDBY state via the COLD_STOP transition. This is all under timer control on a STM32 MCU.

One extra requirement I have is that I need to shutdown a burn at any state by issuing a COLD_STOP command. COLD_STOP command should call the VECU.Stop_ColdBurn(); function.

I have tried using exit states, but I can't get the diagram to generate properly.

What am I doing wrong, what am I missing?

Thanks for any help,
Gronk

**Comment by adamfk** [comment, marked as answer]
Hi Gronk,

This should work. You can see see this in action with example Ex05 https://statesmith.github.io/fundamentals-1/

![image](https://github.com/StateSmith/StateSmith/assets/274012/abcd5c28-6fa0-4b13-a90f-15e61e6f3cf9)

When you say you can't get the diagram to generate properly, do you mean that StateSmith throws an error? Or is the generated code malformed and doesn't compile? Or does it compile and just not do what you want?

We should be able to get this working for you!

Adam

**Reply by Gronk95** [reply]
Thanks, using the Ex05 state diagram solved my problem.

In terms diagrams using exit states not compiling. The diagram below is one example.
![VECU_SM2 drawio](https://github.com/StateSmith/StateSmith/assets/140708711/29162e51-ecf1-4760-95fa-e4e40aabd791)

dotnet-script gives me the following error: "Exit points cannot have triggers".

Thanks,
Gronk

**Reply by adamfk** [reply]
I see. Try making these changes:
```diff
-COLD_STOP via exit cold_stop
+via exit cold_stop
```
Move the `COLD_STOP` trigger to the transition that ends on the red circle exit point.

The main idea here is that on that `COLD_STOP` event, the exit point is entered and it must be exited immediately.
